### DISCUSSION TITLE
How to use protobuf

### DISCUSSION THREAD

**Discussion by jgerits** [vfsfitvnm/frida-il2cpp-bridge#357] [Q&A]
I want to get protobuf data is there a way to easily use protobuf when a method returns a pointer to the object of protobuf?

**Comment by commonuserlol** [comment]
just reverse game and hook method

**Comment by bluewave41** [comment]
There isn't just "a way" to get protobuf data. Protobufs are a concept and can be implemented in any way you wish meaning not all implementations are the same.

You need to dump the game methods using `Il2Cpp.dump()` and then identity the protobuf methods yourself either from searching through the dump or [tracing](https://github.com/vfsfitvnm/frida-il2cpp-bridge/wiki/Snippets#tracing) for calls that might be protobuf related.

That can get you raw schema-less data but the rest is on you to do something with it.

**Reply by jgerits** [reply]
i now get a protobuf back from a call like:


    var x = this.method("GetAllDetails").invoke(a ,b);
    
    x.methods.forEach((method: any) => {
        console.log("method: " + method);
    });
    x.fields.forEach((field: any) => {
        console.log("field: " +field);
    });

but it can't find x.methods in this. If i immediatly call a method in x with x.method("call").invoke(); it says that it does not have that method.
